torch/_dynamo/guards.py

```diff
@@ -972,8 +972,13 @@ def source_ref(source):
             output_graph.global_scope,
             self,
         )
+
+        # Break retain cycle. See test_release_scope_memory
+        def cleanup_builder(b):
+            b.scope = {}
+
         # Break retain cycle. See test_release_input_memory
```

Comment:
this line is now redundant fyi

Reply:
Slightly different line, different cycle

Reply:
oh yeh i can't read